Draft of 1.5 release note. [skip ci] (#7271)
dmlc/xgboost: NEWS.md

XGBoost can perform one-hot encoding based categorical tree splits during model
training. All the other features including prediction, SHAP value computation, feature
importance, and model plotting have support for categorical splits on both CPU and GPU.
Also, all Python interfaces including native interface with and without quantized
``DMatrix``, scikit-learn interface, and Dask interface now accepts categorical data with
a wide range of data structures support including numpy/cupy array and cuDF/pandas/modin
dataframe.  In practice, the following are required for enabling categorical data support
during training:

  - Use Python package.
  - Use ``gpu_hist`` to train the model.
  - Use JSON model file format for saving the model.

Once the model is trained, it can be used with all features that are available on the
Python package.  For a quick demo using the sciki-learn interface, see
https://github.com/dmlc/xgboost/blob/master/demo/guide-python/categorical.py

Related PRs: (#7011, #7001, #7042, #7041, #7047, #7043, #7036, #7054, #7053, #7065, #7213, #7228, #7220, #7221, #7231)

* Next steps

> 	For the coming releases, we will focus on CPU training with categorical data and work
> 	on supporting partition-based tree split instead of one-hot encoding based

Reviewer: ```suggestion
	- Revise the CPU training algorithm to handle categorical data and natively and generate categorical splits
	- Extend the CPU and GPU algorithms to generate categorical splits of form `x \in S` where the input is compared with multiple category values.
```

	split. (#7081)

### External memory

> A new Python interface for external memory based on iterator is introduced in this release

Reviewer: ```suggestion
This release features a brand-new interface and implementation for external memory (also known as out-of-core training).
```

> (#6901, #7064, #7088, #7089, #7087, #7092, #7070, #7216).  The new interface is similar to

Reviewer: ```suggestion
(#6901, #7064, #7088, #7089, #7087, #7092, #7070, #7216). The new implementation leverages the data iterator interface, which is currently used
```

> the one used for ``DeviceQuantileDMatrix`` with ``DataIter``. For a quick introduction,

Reviewer: ```suggestion
to create `DeviceQuantileDMatrix``. For a quick introduction,
```

see https://xgboost.readthedocs.io/en/latest/tutorials/external_memory.html#data-iterator
. During the development of this new interface, ``lz4`` compression is removed. (#7076).
Please note that external memory support is still experimental and not ready for
production use yet.  All future development will focus on this new interface and users are

> advised to migrate from the old data parser in XGBoost.

Reviewer: ```suggestion
advised to migrate. (You are using the old interface if you are using a URL suffix to use external memory.)
```

### New features in Python package

> * Support numpy array interface and all numeric types from numpy in ```DMatrix``

Reviewer: ```suggestion
* Support numpy array interface and all numeric types from numpy in `DMatrix`
```

>   construction and ``inplace_predict`` (#6998, #7003).  Now XGBoost no longer makes data

Reviewer: ```suggestion
  construction and `inplace_predict` (#6998, #7003).  Now XGBoost no longer makes data
```

  copy when input is numpy array.
* The early stopping callback in Python has a new ``min_delta`` parameter to control the
  stopping behavior (#7137)
* Python package now supports calculating feature scores for the linear model, which is
  also available on R package. (#7048)
* Python interface now supports configuring constraints using feature names instead of
  feature indices.
* Typehint support for more Python code including scikit-learn interface and rabit
  module. (#6799, #7240)
Add tutorial for XGBoost-Ray (#6884)

### New features in R package

> * In 1.4 we have a new prediction C API and is used in Python.  This release adds support
>   for the R package.  A new parameter ``iteration_range`` for the predict function is

Reviewer: ```suggestion
* In 1.4 we have a new prediction function in the C API which is used by the Python package.  This release revises
  the R package to use the new prediction function as well.  A new parameter ``iteration_range`` for the predict function is
```

* R package now supports the ``nthread`` parameter in ``DMatrix`` construction. (#7127)

### New features in JVM packages
* Support GPU dataframe and ``DeviceQuantileDMatrix`` (#7195).  Constructing ``DMatrix``
  with GPU data structures and the interface for quantized ``DMatrix`` were first
  introduced in the Python package and are now available in the xgboost4j package.
* JVM packages now support saving and getting early stopping attributes. (#7095) Here is a
  quick [example](https://github.com/dmlc/xgboost/jvm-packages/xgboost4j-example/src/main/java/ml/dmlc/xgboost4j/java/example/EarlyStopping.java "example") in JAVA (#7252).

### General new features
* We now have a pre-built binary package for R on Windows with GPU support. (#7185)
* CUDA compute capability 86 is now part of the default CMake build configuration with
  newly added support for CUDA 11.4. (#7131, #7182, #7254)
* XGBoost can be compiled using system CUB provided by CUDA 11.x installation. (#7232)

### Optimizations
The performance for both ``hist`` and ``gpu_hist`` has been significantly improved in 1.5
with the following optimizations:
* GPU multi-class model training now supports prediction cache. (#6860)

> * GPU histogram building is speedup and the overall training time is 2~3 times faster on

Reviewer: ```suggestion
* GPU histogram building is sped up and the overall training time is 2-3 times faster on
```

>   large datasets (#7180, #7198).  During development ``deterministic_histogram`` is remove
>   and the GPU algorithm is always deterministic.

Reviewer: ```suggestion
  large datasets (#7180, #7198).  In addition, we removed the parameter `deterministic_histogram` and now
  the GPU algorithm is always deterministic.
```

* CPU hist has an optimized procedure for data sampling (#6922)

> * More performance is extracted from regression and binary classification objectives on

Reviewer: ```suggestion
* More performance optimization in regression and binary classification objectives on
```
